cpp/src/sort/common_sort_impl.cuh

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/table/table_view.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/traits.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/sort.h>
+
+namespace cudf {
+namespace detail {
+
+/**
+ * @brief The enum specifying which sorting method to use (stable or unstable).
+ */
+enum class sort_method { STABLE, UNSTABLE };
+
+/**
+ * @brief Fast-path sort a column in place
+ *
+ * Precondition, is_usable(column) returned true
+ *
+ * @tparam method Whether to use a stable sort or not.
+ * @param col Column to sort, modified in place.
+ * @param order Ascending or descending sort order.
+ * @param stream CUDA stream used for device memory operations and kernel launches
```

Comment:
These should be with the `operator()` function instead of the class doxygen.

Reply:
Done, thanks.